refinery/tool_manager/management/commands/generate_tool_definitions.py

```diff
@@ -32,109 +33,122 @@ class Command(BaseCommand):
             dest='workflows',
             help='Generate ToolDefinitions for properly annotated '
                  'Galaxy-based Workflows'
+        ),
+        make_option(
+            '--force',
+            action='store_true',
+            dest='force',
+            help='Overwrite ToolDefinitions with the same names instead of '
+                 'skipping their creation'
         )
     )
 
+    def __init__(self):
+        super(Command, self).__init__()
+        self.force = False
+
     def handle(self, *args, **options):
         """
         Creates ToolDefinitions based off of properly annotated Galaxy
         Workflows and Visualization Tools.
         """
+        self.force = options["force"]
+
+        if self.force:
+            self._confirmation_loop()
 
         if not options["visualizations"] and not options["workflows"]:
-            self.generate_visualization_tool_definitions()
-            self.generate_workflow_tool_definitions()
+            self.generate_tool_definitions()
 
         if options["visualizations"]:
-            self.generate_visualization_tool_definitions()
-
+            self.generate_tool_definitions(workflows=False)
         if options["workflows"]:
-            self.generate_workflow_tool_definitions()
-
-    def generate_visualization_tool_definitions(self):
-        """
-        Validate visualization annotation data, and try to create a
-        ToolDefinition if our validation rules pass.
+            self.generate_tool_definitions(visualizations=False)
+
+    @staticmethod
+    def _ask_for_confirmation():
+        return raw_input("Are you sure you want to `--force`? This will "
+                         "delete any existing ToolDefinitions with the "
+                         "same name as any new ones you you're trying to "
+                         "import: [y/n]: ")
+
+    def _check_for_duplicates(self, tool_annotation):
+        current_tool_definition_names = [
+            t.name for t in ToolDefinition.objects.all()
+        ]
+        if tool_annotation["name"] in current_tool_definition_names:
+            if self.force:
+                self.stdout.write(
+                    self.style.NOTICE("Forcing deletion of of `{}`".format(
+                            tool_annotation["name"]))
+                )
+                ToolDefinition.objects.get(
+                    name=tool_annotation["name"]
+                ).delete()
```

Comment:
Should there be a `return False` here? If the callers are depending on a false-y value, better to be explicit: I'd worry abut the `return True` being outdented at some point.

Or I could be confused.

Reply:
Nope that make total sense, I'll add thanks!

Reply:
or maybe retitle to `has_duplicates`?